pallets/phala/src/stakepool.rs

```diff
@@ -540,6 +562,110 @@ pub mod pallet {
 
 			Ok(())
 		}
+		/// add a staker accountid to contribution whitelist.
+		/// calling this method will forbide stakers contribute who isn't in the whitelist.
+		/// caller must be the owner of the pool.
+		/// if a pool hasn't registed in the wihtelist map, any staker could contribute as what they use to do.
+		/// whitelist has a lmit len of 500 stakers.
+		#[pallet::weight(0)]
+		pub fn add_staker_to_whitelist(
+			origin: OriginFor<T>,
+			pid: u64,
+			staker: T::AccountId,
+		) -> DispatchResult {
+			let owner = ensure_signed(origin)?;
+			let mut pool_info = Self::ensure_pool(pid)?;
+			ensure!(pool_info.owner == owner, Error::<T>::UnauthorizedPoolOwner);
+			if let Some(mut whitelist) = LimitContribtionPoolsList::<T>::get(&pid) {
+				if !whitelist.contains(&staker) {
+					ensure!(whitelist.len() as u32 <= MAX_WHITELIST_LEN, Error::<T>::ExceedWhitelistMaxLen);
```

Comment:
In this way, you allow whitelist to have MAX+1 elements

Reply:
done

Reply:
Not fixed. You first check the whitelist len, and push one element later, resulting it can have 101 elements